src/api/app/models/concerns/staging_project.rb

```diff
@@ -87,19 +85,32 @@ def broken_packages
     @broken_packages
   end
 
+  def missing_reviews_for_classified_requests(request, reviews)
+    @missing_reviews_of_st_project ||= []
+    current_missing_reviews = []
+
+    reviews.each do |review|
```

Comment:
You can use a `reduce` here, capture the returning accumulator and then append it to `@missing_reviews_of_st_project`. Doing so you don't need the `current_missing_reviews` variable outside and the double append inside the loop

Reply:
@danidoni thanks for the hint, I will have a look :)

Reply:
@danidoni the ```extract_missing_reviews``` returns an hash, the ```reduce``` would convert the hash to an array. Do I see it correct? Do you have something in mind that leads to the expected result :)

Reply:
@krauselukas hm but maybe you can use the advice from @DavidKang here again and  write the `next if review.state == :accepted || review.by_project == name` as a `where.not(...)` condition ?

Reply:
@vpereira Check again, I filter the reviews now in the database query. It was a bit more tricky than I thought in the beginning. Using ```where.not``` for the ```by_project``` is a problem if it is nil. But by using ```or``` I was able to get the expected results.

Reply:
@danidoni I'm using the ```reduce``` now :)